src/ai/backend/manager/models/group.py

```diff
@@ -1148,3 +1176,25 @@ async def get_projects(
             permissions = await permission_ctx.calculate_final_permission(row)
             result.append(ProjectModel.from_row(row, permissions))
     return result
+
+
+async def get_permission_ctx(
+    db_conn: SAConnection,
+    ctx: ClientContext,
+    requested_permission: ProjectPermission,
+    target_scope: Optional[ScopeType] = None,
+    container_registry_scope: Optional[ContainerRegistryScope] = None,
+) -> ProjectPermissionContext:
+    async with ctx.db.begin_readonly_session(db_conn) as db_session:
+        builder = ProjectPermissionContextBuilder(db_session)
+
+        if target_scope is not None:
+            permission_ctx = await builder.build(ctx, target_scope, requested_permission)
+        elif container_registry_scope is not None:
+            permission_ctx = await builder.build_ctx_in_container_registry_scope(
+                ctx, container_registry_scope
+            )
+        else:
+            raise ValueError("either target_scope or container_registry_scope must be provided")
+
```

Comment:
In our RBAC system, target scope should not be null. You can use `SystemScope` if you need to query in a system wide scope.